typed/merge.go

```diff
@@ -67,11 +71,11 @@ var (
 func (w *mergingWalker) merge() (errs ValidationErrors) {
 	if w.lhs == nil && w.rhs == nil {
 		// check this condidition here instead of everywhere below.
-		return w.errorf("at least one of lhs and rhs must be provided")
+		return errorf("at least one of lhs and rhs must be provided")
```

Comment:
How can a caller know the path that this error came from?

Reply:
I think this error would only happen at root anyway, since I can't think of a way that we would create a mergingWalker with no lhs or rhs sides from descending.

Reply:
See rest of the discussion about `WithPrefix`